libgit2-sys/lib.rs

```diff
@@ -1728,6 +1729,35 @@ pub struct git_cherrypick_options {
     pub checkout_opts: git_checkout_options,
 }
 
+pub type git_apply_delta_cb =
+    Option<extern "C" fn(delta: *const git_diff_delta, payload: *mut c_void) -> c_int>;
+
+pub type git_apply_hunk_cb =
+    Option<extern "C" fn(hunk: *const git_diff_hunk, payload: *mut c_void) -> c_int>;
+
+git_enum! {
+    pub enum git_apply_flags_t {
+        GIT_APPLY_CHECK = 1<<0,
+    }
+}
+
+#[repr(C)]
+pub struct git_apply_options {
+    pub version: c_uint,
+    pub delta_cb: git_apply_delta_cb,
+    pub hunk_cb: git_apply_hunk_cb,
+    pub payload: *mut c_void,
+    pub flags: u32,
```

Comment:
Hey @alexcrichton thanks for merging!

Out of curiosity: why is it not possible to idiomatically use the enum types in the structs? This only seemed to fail on windows:
https://github.com/rust-lang/git2-rs/runs/521702373

I see many structs using the `u32` instead of the actual enum type, but some seem to be able to use them, what am I missing?

Reply:
I think it has to do with the way MSVC handles enums and whether they're signed or not. I don't remember a ton of specifics but I know it's slightly different from gcc which has caused issues. In any case I'm fine with tweaking things as necessary!